### PR TITLE
Pack targets

### PR DESCRIPTION
**Pull request by c1728p9**
Add support for targets in cmsis-packs.

Note - This PR is blocked until arm-pack-manager is released. It has been pushed up for review only.

### CODE REVIEW

File: pyOCD/test/test_target/pack/test_pack_target.py [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+    args_and_results = [
+        ((b"",), []),
+        ((b"a",), [0x61]),
+        ((b"abc",), [0x636261]),
+        ((b"abcd",), [0x64636261]),
+        ((b"abcde",), [0x64636261, 0x65]),
+        ((bytearray(b"abcde"),), [0x64636261, 0x65]),
+    ]
+    for args, result in args_and_results:
+        assert list(result) == list(_bytes_to_words(*args))
```
Comment (theotherjimmy) [inline pyOCD/test/test_target/pack/test_pack_target.py]: Could you add
```python
assert list(result) == list(_bytes_to_words(bytearray(args))
```
and then have a bytearray test for all of them?

**Comment by pfried** [comment]
Is there any progress on this? I would really like to get debugging working for some of our Silicon Labs boards and this may help

**Comment by c1728p9** [comment]
Hi @pfried this is blocked on the back end package management tool. CC @theotherjimmy

**Comment by theotherjimmy** [comment]
@c1728p9 I'll make that top priority.

**Comment by Hoohaha** [comment]
Hello @c1728p9 ,  I think this is a great feature for supporting cmsis packs. The cmsis-pack-manager is not necessary, user can maunal download from sites or use their offline packs, if automated cache is done that' would be more better. How do you think? If this feature is enabled, the pyocd could solve many problems for us.

**Comment by c1728p9** [comment]
Closing this to reduce clutter. @flit feel free to take anything in this PR that is useful.

**Pull request closed without merging**